packages/editor-ui/src/composables/useCanvasMapping.ts

```diff
@@ -534,10 +535,15 @@ export function useCanvasMapping({
 			} else if (runDataTotal > 0) {
 				status = 'success';
 			}
+
+			maxConnections = [NodeConnectionType.Main, NodeConnectionType.AiTool].includes(type)
```

Comment:
Maybe instead of relying on type you could use `maxConnections`? It's defined on a connection in the node type

<img width="681" alt="image" src="https://github.com/user-attachments/assets/8536b0c2-21d1-42a4-afc1-df72f08a574b">